### PR TITLE
Bug 1112324 - Privacy Day home page promo

### PR DESCRIPTION
**Pull request by craigcook**
Waffle switch: "mozorg-home-privacy-promo" to activate on January 28 and deactivate on February 4.
Waffle switch: "mozorg-home-privacy-promo-jan28" to activate on January 28 and deactivate on January 29.

### CODE REVIEW

File: bedrock/mozorg/templates/mozorg/home/home-new.html [hunk 1]
```diff
@@ -136,6 +144,27 @@ <h2 class="go">{{ _('Download Firefox') }}</h2>
             <a class="fxos-link" href="{{ url('firefox.desktop.index') }}">{{ _('Learn more') }}</a>
           </div>
         </li>
+      {% if waffle.switch('mozorg-home-privacy-day') and LANG.startswith('en') %}
+        <li id="promo-6" class="item promo-large-landscape privacy-day" tabindex="0" data-name="January 28 is Data Privacy Day">
+          <h2 class="primary go">{% if LANG == 'en-US' %}January 28{% else %}28 January{% endif %} is Data Privacy Day</h2>
```
Comment (craigcook) [inline bedrock/mozorg/templates/mozorg/home/home-new.html]: I may be overthinking this bit. We don't localize this page for en-ZA or en-GB do we?

**Comment by craigcook** [comment]
attn @flodolo, here be strings

File: bedrock/mozorg/templates/mozorg/home/home-new.html (hunk 1 above)
Comment (flodolo) [inline bedrock/mozorg/templates/mozorg/home/home-new.html]: We're working to have en-GB, we don't have en-ZA. For this specific one we probably can target en-*

File: bedrock/mozorg/templates/mozorg/home/home-new.html [hunk 2]
```diff
@@ -115,11 +115,19 @@ <h2 class="primary go">{{ _('Discover <br>what drives our mission') }}</h2>
           </a>
         </li>
       {% endif %}
+      {% if l10n_has_tag('privacy_day') %}
+        <li id="promo-2" class="item promo-small-landscape advocacy" data-name="Learn about Mozilla Advocacy">
+          <a class="panel-link" href="https://advocacy.mozilla.org">
+            <h2>Learn about Mozilla Advocacy</h2>
```
Comment (flodolo) [inline bedrock/mozorg/templates/mozorg/home/home-new.html]: Not wrapped

Comment (craigcook) [inline bedrock/mozorg/templates/mozorg/home/home-new.html]: Oops! Fixed.

File: bedrock/mozorg/templates/mozorg/home/home-new.html [hunk 3]
```diff
@@ -115,11 +115,19 @@ <h2 class="primary go">{{ _('Discover <br>what drives our mission') }}</h2>
           </a>
         </li>
       {% endif %}
+      {% if waffle.switch('mozorg-home-privacy-promo') and l10n_has_tag('privacy_day') %}
```
Comment (jgmize) [inline bedrock/mozorg/templates/mozorg/home/home-new.html]: micro-optimization nit: put the l10n_has_tag check in front of the waffle.switch to short-circuit a potentially unnecessary DB query.

File: bedrock/mozorg/templates/mozorg/home/home-new.html [hunk 4]
```diff
@@ -136,6 +144,29 @@ <h2 class="go">{{ _('Download Firefox') }}</h2>
             <a class="fxos-link" href="{{ url('firefox.desktop.index') }}">{{ _('Learn more') }}</a>
           </div>
         </li>
+    {% if waffle.switch('mozorg-home-privacy-promo') and l10n_has_tag('privacy_day') %}
+      {% if waffle.switch('mozorg-home-privacy-promo-jan28') and LANG.startswith('en') %}
```
Comment (jgmize) [inline bedrock/mozorg/templates/mozorg/home/home-new.html]: I don't see why you need to nest these. The `mozorg-home-privacy-promo-jan28` switch shouldn't ever be active independent of the `mozorg-home-privacy-promo` switch, right? Also, you're checking both l10n_has_tag and LANG.startswith('en'), which is also redundant. I would just use the inner conditional and get rid of the outer one.
```

Comment (jgmize) [inline bedrock/mozorg/templates/mozorg/home/home-new.html]: ah, I missed the outer `else`. can't get rid of it, never mind.

Comment (jgmize) [inline bedrock/mozorg/templates/mozorg/home/home-new.html]: Same micro-optimization nits as above: move the waffle.switch checks after the l10n_has_tag and LANG.startswith checks.

File: media/css/mozorg/home/home-promo.less [hunk 5]
```diff
@@ -1656,6 +1685,12 @@ html[lang|="en"] {
         .background-size(121px, 120px);
     }
 
+    // Advocacy
+    .promo-small-landscape.advocacy {
+        background-image: url(/media/img/home/voices/promos/advocacy/advocacy-high-res.png);
```
Comment (alexgibson) [inline media/css/mozorg/home/home-promo.less]: should this be `advocacy-logo-high-res.png`?

**Comment by alexgibson** [comment]
Aside from the file name typo, if @jgmize is happy with the waffle promo inceptioness, this is an r+wc from me :banana:

**Comment by craigcook** [comment]
Yeah I was iffy on the waffleception. It's a bit kludgy but was the best approach I could come up with apart from duplicating code. There's one switch to toggle the whole promo (if it's translated), and within that a more specific switch to toggle the January 28 state of that promo (only for English).

**Pull request merged**